lldb/source/Target/Target.cpp

```diff
@@ -1488,7 +1488,7 @@ static void LoadTypeSummariesForModule(ModuleSP module_sp) {
         auto summary_sp =
             std::make_shared<StringSummaryFormat>(flags, summary_string.data());
         FormatterMatchType match_type = eFormatterMatchExact;
-        if (summary_string.front() == '^' && summary_string.back() == '$')
+        if (type_name.front() == '^')
```

Comment:
It just dawned on me that in Pascal, `^` is the equivalent of `*` in C and thus can be part of a type name. In the next revision we probably should include an isRegex bit in the header.

Reply:
and AFAIK there are downstream Pascal compilers that target LLDB.